setup.py

```diff
@@ -11,6 +11,9 @@
 SRC_DIR = os.path.join(THIS_DIR, "src")
 SNOWPARK_SRC_DIR = os.path.join(SRC_DIR, "snowflake", "snowpark")
 CONNECTOR_DEPENDENCY_VERSION = ">=2.7.12, <4.0.0"
+REQUIRED_PYTHON_VERSION = "==3.8.*"
+if os.getenv("SNOWFLAKE_IS_PYTHON_RUNTIME_TEST", False):
+    REQUIRED_PYTHON_VERSION = ">=3.8"
```

Comment:
should we do `>=3.8, <=3.9`

Reply:
I was debating about adding `<3.10` but then thought that `SNOWFLAKE_IS_PYTHON_RUNTIME_TEST` indicates that we are testing unsupported python versions, and we probably don't need to limit the version.

This way we don't need to change this `REQUIRED_PYTHON_VERSION` again for `3.10` testing. What do you think?

Reply:
I prefer adding `<3.10` because it tells users that you can't use 3.10. Or else somebody will wrongly install snowpark with python 3.10 successfully then found it didn't work with UDF.

Reply:
Hmm but `SNOWFLAKE_IS_PYTHON_RUNTIME_TEST` is for testing only, and it also does not work with 3.9, right?